spec/haml_lint/cli_spec.rb

```diff
@@ -38,6 +38,27 @@
       end
     end
 
+    context 'when a logger is not passed' do
+      let(:logger) { nil }
+
+      it 'sets a default logger that writes to stdout' do
+        $stdout = StringIO.new
+        subject
+        expect(cli.send(:log).instance_variable_get(:'@out')).to eq($stdout)
+      end
+    end
+
+    context 'when passed the --stderr flag' do
+      let(:logger) { nil }
+      let(:args) { %w[--stderr] }
+
+      it 'sets a logger that writes to stderr' do
+        $stderr = StringIO.new
+        subject
+        expect(cli.send(:log).instance_variable_get(:'@out')).to eq($stderr)
+      end
+    end
```

Comment:
These are preeeetty gross, but they're as direct as I could make them without exposing parts of `CLI` and `Logger` that don't need to be exposed.

The `$stdout/err = StringIO.new` keeps these test runs from cluttering up the rspec output.